src/self-managed-debug/src/main.rs

```diff
@@ -0,0 +1,722 @@
+        .list(&ListParams::default())
+        .await
+    {
+        Ok(list) => list.items,
+        Err(e) => {
+            eprintln!(
+                "Failed to get statefulsets for namespace {}: {}",
+                namespace, e
+            );
+            Vec::new()
+        }
+    };
+
+    let network_policies = match Api::<NetworkPolicy>::namespaced(client.clone(), namespace)
+        .list(&ListParams::default())
+        .await
+    {
+        Ok(list) => list.items,
+        Err(e) => {
+            eprintln!(
+                "Failed to get network policies for namespace {}: {}",
+                namespace, e
+            );
+            Vec::new()
+        }
+    };
+
+    enum FormatReadyRatioInput<'a> {
+        Statuses(&'a Option<Vec<ContainerStatus>>),
+        ReadyCount((i32, i32)), // (ready_count, total_count)
+    }
+
+    let format_ready_ratio = |input: FormatReadyRatioInput| match input {
+        FormatReadyRatioInput::Statuses(container_statuses) => match container_statuses {
+            Some(statuses) => format!(
+                "{}/{}",
+                statuses.iter().filter(|c| c.ready).count(),
+                statuses.len()
+            ),
+            None => "N/A".to_string(),
+        },
+        FormatReadyRatioInput::ReadyCount((ready, total)) => format!("{}/{}", ready, total),
+    };
+
+    let format_restart_count =
+        |container_statuses: &Option<Vec<ContainerStatus>>| match container_statuses {
+            Some(statuses) => statuses.iter().map(|c| c.restart_count).sum(),
+            None => 0,
+        };
+
+    let format_age = |metadata: &ObjectMeta| {
+        let start_time = metadata.creation_timestamp.clone();
+        if let Some(start_time) = start_time {
+            let duration: Duration = Utc::now() - start_time.0;
+            format_duration(duration)
+        } else {
+            "N/A".to_string()
+        }
+    };
+
+    // Pods
+    let pod_info: Vec<PodInfo> = pods
+        .into_iter()
+        .filter_map(|pod| {
+            pod.metadata.name.clone().map(|name| {
+                let status = pod.status.unwrap_or_default();
+                PodInfo {
+                    name,
+                    ready: format_ready_ratio(FormatReadyRatioInput::Statuses(
+                        &status.container_statuses,
+                    )),
+                    status: status.phase.unwrap_or_default(),
+                    restarts: format_restart_count(&status.container_statuses),
+                    age: format_age(&pod.metadata),
+                    ip: status.pod_ip.unwrap_or_default(),
+                    node: pod.spec.unwrap_or_default().node_name.unwrap_or_default(),
+                }
+            })
+        })
+        .collect();
+
+    if !pod_info.is_empty() {
+        writeln!(
+            file,
+            "Pods:\n{}\n",
+            Table::new(pod_info).with(Style::psql())
+        )?;
```

Comment:
These structs might be useful for printing things out for a human running the tool, but I really just want the yaml serialized object in its entirety.